app/src/main/java/com/example/feelgoodinc/services/CalendarService.java

```diff
@@ -0,0 +1,265 @@
+package com.example.feelgoodinc.services;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.ServiceConnection;
+import android.graphics.Color;
+import android.os.IBinder;
+
+import androidx.core.content.ContextCompat;
+
+import com.example.feelgoodinc.R;
+import com.example.feelgoodinc.models.Journal;
+import com.example.feelgoodinc.models.Mood;
+import com.github.sundeepk.compactcalendarview.CompactCalendarView;
+import com.github.sundeepk.compactcalendarview.domain.Event;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * <p>Uses the {@link JournalService} and {@link MoodService} to retrieve the {@link Journal} and {@link Mood} of the month</p>
+ *
+ */
+public class CalendarService {
+    private final CalendarServiceCallback callback;
+    public MoodService moodService;
+    public JournalService journalService;
+    Context context;
+    public boolean journalBound;
+    public boolean moodBound;
+
+    /**
+     * Creates the calendar service object
+     * @param context the context of the app
+     */
+    public CalendarService(Context context,CalendarServiceCallback callback){
+        this.callback = callback;
+        this.context = context;
+        bindToJournalService();
```

Comment:
Interesting idea to bind a `Service` in another `Service. Thought it was bad practice, but apparently it's what's supposed to happen.